src/components/Calculator.js

```diff
@@ -1,111 +1,134 @@
 /* eslint-disable no-unused-vars */
 import React from 'react';
 import './UI.css';
+import calculate from './logic/Calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
-    super();
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      // operation: null,
+    };
+  }
+
+  handleClick = (event) => {
+    const { value } = event.target;
+    this.setState((prevState) => calculate(prevState, value));
   }
 
   render() {
+    const { total, next } = this.state;
+    const displayValue = next || total || '0';
     return (
       <div className="calculator">
-        <div className="calculator__display">0</div>
+        <div className="calculator__display">{displayValue}</div>
         <div className="calculator__keys">
           <button
             type="submit"
             className="calculator__key calculator__key--operator"
             value="AC"
+            onClick={this.handleClick}
           >
             AC
           </button>
           <button
             type="submit"
             className="calculator__key calculator__key--operator"
             value="+/-"
+            onClick={this.handleClick}
           >
             +/-
           </button>
           <button
```

Comment:
- [ ] As per the proct requirements [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_events.md#:~:text=Note%3A%20Practice%20what%20you%20have%20learned%20until%20now%3A%20use%20useState%20and%20props.) kindly make sure to use props in your project by using child components.

__Hints: for this you might have to create a component button that you will use everywhere you need button, giving it the handleclick function and the right value as props __